src/ProfileEndpoints.jl

```diff
@@ -242,6 +245,30 @@ end
 
 end  # if isdefined
 
+###
+### Type Inference
+###
+
+function typeinf_start_endpoint(req::HTTP.Request)
+    Core.Compiler.__set_measure_typeinf(true)
+    return HTTP.Response(200, "Type inference profiling started.")
+end
+
+function typeinf_stop_endpoint(req::HTTP.Request)
+    Core.Compiler.__set_measure_typeinf(true)
+    timings = if isdefined(Core.Compiler.Timings, :clear_and_fetch_timings)
+        # TODO: make a root node out of this
+        Core.Compiler.Timings.clear_and_fetch_timings()
```

Comment:
I'd like to just spoof an `InferenceTimingNode` here instead of changing `flamegraph` downstream to work with the array of trees, if possible

Reply:
Yeah, i think that's the right approach. i ended up doing that too in the snoopcompile PR, here:
https://github.com/timholy/SnoopCompile.jl/pull/313#discussion_r1059492749